--- oaf-0.6.5/api-docs/Makefile.am~	Thu Nov 16 10:14:56 2000
+++ oaf-0.6.5/api-docs/Makefile.am	Thu Jun 14 18:31:51 2001
@@ -81,6 +81,7 @@
 maintainer-clean-local: clean
 	cd $(top_builddir)/api-docs/ && rm -rf sgml html $(DOC_MODULE)-decl-list.txt $(DOC_MODULE)-decl.txt
 
+if ENABLE_GTK_DOC
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(TARGET_DIR)
 	(installfiles=`echo $(srcdir)/html/*.html`; \
@@ -96,6 +97,7 @@
 	  echo '-- Fixing Crossreferences' ; \
 	  gtkdoc-fixxref --module=$(DOC_MODULE) --html-dir=$(HTML_DIR)|| true; \
 	fi)
+endif
 
 dist-hook: sgml html
 	mkdir $(distdir)/html
--- oaf-0.6.5/api-docs/Makefile.am~	Thu Jun 14 18:35:57 2001
+++ oaf-0.6.5/api-docs/Makefile.am	Thu Jun 14 18:37:44 2001
@@ -23,10 +23,6 @@
 
 
 liboaf_docdir = $(HTML_DIR)
-liboaf_doc_DATA =			\
-	liboaf.html			\
-	$(NULL)
-
 
 my_sgml_files = 			\
 	tutorial.sgml			\
